lib/ansible/modules/cloud/amazon/aws_caller_facts.py

```diff
@@ -71,15 +76,30 @@ def main():
     client = module.client('sts')
 
     try:
-        caller_identity = client.get_caller_identity()
-        caller_identity.pop('ResponseMetadata', None)
-        module.exit_json(
-            changed=False,
-            **camel_dict_to_snake_dict(caller_identity)
-        )
+        caller_facts = client.get_caller_identity()
+        caller_facts.pop('ResponseMetadata', None)
     except (BotoCoreError, ClientError) as e:
         module.fail_json_aws(e, msg='Failed to retrieve caller identity')
 
+    iam_client = module.client('iam')
+    alias = ''
+
+    try:
+        # Although a list is returned by list_account_aliases AWS supports maximum one alias per account.
+        # If an alias is defined it will be returned otherwise a blank string is filled in as account_alias.
+        # see https://docs.aws.amazon.com/cli/latest/reference/iam/list-account-aliases.html#output
+        response = iam_client.list_account_aliases()
```

Comment:
I think if this fails, we can safely return `None` or `''` as the account alias, because it's possible people don't have the additional ListAccountAliases permission. That would be backwards compatible as well, since setups that already work without that permission would suddenly break in this version.

Reply:
I like the idea of retaining backwards compatability on the permissons, but I'd like to maintain a distinction between being unable to get the account alias and there being no account alias.

How about the following:

* If we cannot get the account alias because of permissions of other failure: Do not include the account_alias key in the response
* If we are able to make the call and get the result of no account alias: Include account_alise = None in the response
* If we succeed in getting the account alias: Include account_alias = _the alias_ in the response

Reply:
I'm fine with the latest suggestion.
I would like to add a pointer in the documentation on this behavior too.
And possibly raise a warning?
What about the test case btw, how should that one behave?

Reply:
I'd be ok with skipping an integration test for this, and adding the CI perms so the happy path (when the module does have permissions) is tested. Agreed w/ @orthanc about not adding a warning.